src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1437,7 +1435,17 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
                    << "' not found among registered applications.");
       return;
     }
-    if (is_allowed) {
+
+    bool is_allowed_by_policies = true;
+    if (PolicyEnabled()) {
+      std::string policy_app_id = app->policy_app_id();
+      AppPermissions permissions(policy_app_id);
+      permissions = policy_manager->GetAppPermissionsChanges(app->mac_address(),
+                                                             policy_app_id);
+      is_allowed_by_policies = !permissions.appRevoked;
```

Comment:
```suggestion
      is_allowed_by_policies =
          policy_manager->IsApplicationRevoked(app->policy_app_id());
```
Seems the current fix has some obscure issues, try this instead

Reply:
Fixed in https://github.com/smartdevicelink/sdl_core/pull/3592/commits/842f85ca1ee94fbfb5abc40de8e2be42ac5a1479